Sources/TuistKit/Generator/ProjectGenerator.swift

```diff
@@ -184,12 +191,15 @@ final class ProjectGenerator: ProjectGenerating {
         let generatedProject = GeneratedProject(path: xcodeprojPath,
                                                 targets: nativeTargets)
 
-        try writeXcodeproj(workspace: workspace,
-                           pbxproj: pbxproj,
-                           xcodeprojPath: xcodeprojPath)
-        try writeSchemes(project: project,
-                         generatedProject: generatedProject,
-                         graph: graph)
+        try fileHandler.inTemporaryDirectory { temporaryPath in
+            try writeXcodeproj(workspace: workspace,
+                               pbxproj: pbxproj,
+                               xcodeprojPath: temporaryPath)
+            try writeSchemes(project: project,
+                             generatedProject: generatedProject,
+                             graph: graph)
+            try fileHandler.replace(xcodeprojPath, with: temporaryPath)
```

Comment:
yikes - looks like the schemes aren't generated in the temporary directory (rather they are generated in the final destination) and as a result get wiped when we do a replace - luckily the acceptance tests caught this! :)

Reply:
True. Glad that we are investing into having a good suite of acceptance tests.